### PR TITLE
[spirv] Allow performing scalar load/store when promotion

### PR DESCRIPTION
**Pull request by antiagainst**
This enables us to have software pipelining and multibuffering as the next step.

### CODE REVIEW

**Comment by antiagainst** [comment]
Depends on https://reviews.llvm.org/D137471.

File: compiler/src/iree/compiler/Codegen/Passes.h [hunk 1]
```diff
@@ -521,6 +521,10 @@ std::unique_ptr<OperationPass<func::FuncOp>> createSPIRVVectorizePass();
 /// having pointer bitcast.
 std::unique_ptr<OperationPass<ModuleOp>> createSPIRVVectorizeLoadStore();
 
+/// Breaks down large vendors not natively supported by SPIR-V.
```
Comment (kuhar) [inline compiler/src/iree/compiler/Codegen/Passes.h:524]: s/vendors/vectors/

**Review by kuhar — APPROVED** [review]
LGTM

File: compiler/src/iree/compiler/Codegen/Passes.h (hunk 1 above)
Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/Codegen/Passes.h:524]: could we break down large vendors that dont natively support SPIR-V. That would be really good result! :D

Comment (antiagainst) [reply compiler/src/iree/compiler/Codegen/Passes.h]: lol. Silly typo by me. Fixed.

**Comment by antiagainst** [comment]
Needs https://github.com/iree-org/iree/pull/11111.

**Pull request merged**